values.go

```diff
@@ -124,6 +126,7 @@ func (value *Number) Type() Type {
 	return value.typ
 }
 
+// usual number equality requires that both value and scale are equal
```

Comment:
which isn't what we want in the language when doing `==` because we'd very likely do `a != 0 && b / a` and would not want that to be a division by zero

we should fix equality separately to focus on numerical equality

on style: comment in Go should start by the function name e.g. `Equal compares two ....`